### PR TITLE
NMA-6100: New SatsSearchBar component

### PR DESCRIPTION
**Pull request by cecilieopperudsats**
Inspired by the material 3 [Search Bar Component ](https://m3.material.io/components/search/guidelines#3b162db3-8d55-425a-920b-95b1041ff999) ✨ 

image

### CODE REVIEW

File: sats-dna/src/main/kotlin/com/sats/dna/components/SatsSearchBar.kt [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+import androidx.compose.ui.text.input.KeyboardCapitalization
+import androidx.compose.ui.unit.dp
+import com.sats.dna.components.TrailingContent.Icon
+import com.sats.dna.components.button.SatsButtonColor
+import com.sats.dna.components.button.SatsIconButton
+import com.sats.dna.theme.SatsTheme
+import com.sats.dna.tooling.LightDarkPreview
+
+@Composable
+fun M3SatsSearchBar(
```
Comment (sindrenm) [inline sats-dna/src/main/kotlin/com/sats/dna/components/SatsSearchBar.kt:31]: Wanna just call it `SatsSearchBar`? Soon we'll probably be ready to ditch Material 2 altogether. I hope. 😁

File: sats-dna/src/main/kotlin/com/sats/dna/components/SatsSearchBar.kt [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    query: String,
+    onQueryChange: (query: String) -> Unit,
+    onClearClicked: () -> Unit,
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+    keyboardActions: KeyboardActions = KeyboardActions.Default,
+    placeholder: @Composable (() -> Unit)? = null,
```
Comment (sindrenm) [inline sats-dna/src/main/kotlin/com/sats/dna/components/SatsSearchBar.kt:38]: Maybe it's cleaner to just allow a `String?` for the placeholder? Then we create the `Text()` internally with the correct styles. Alternatively, we can wrap the usages of the `placeholder: @Composable () -> Unit` inside of a `ProvideTextStyle {}`.

**Review by sindrenm — APPROVED** [review]
Sweet Christmas!

File: sats-dna/src/main/kotlin/com/sats/dna/components/SatsSearchBar.kt (hunk 2 above)
Comment (cecilieopperudsats) [reply sats-dna/src/main/kotlin/com/sats/dna/components/SatsSearchBar.kt]: True

**Pull request merged**